Hackney.Core/Hackney.Core.Validation.AspNet/ReadMe.md

```diff
@@ -6,24 +6,54 @@ Fluent Validation allows you to separate validation rules from your object model
 
 #### UseErroCodeInterceptor
 
-We have customised our FluentValidation through the use of an interceptor. The `UseErrorCodeInterceptor` implements the `IValidatorInterceptor`.
+This class is used to configure error code and error messages that would be provided when a validation error occurs. This allows the client to understand the validation error that has occured and was of resolving it. 
 
-This interface has two methods – `BeforeAspNetValidation` and `AfterAspNetValidation`.
-
-`BeforeMvcValidation` is invoked after the appropriate validator has been selected but before it is invoked. One of the arguments passed to this method is a `ValidationContext` that will eventually be passed to the validator. The context has several properties including a reference to the object being validated. If we want to change which rules are going to be invoked (for example, by using a custom `ValidatorSelector`) then we can create a new `ValidationContext`, set its Selector property, and return that from the `BeforeAspNetValidation` method.
+Usage
+```csharp
 
-Likewise, `AfterAspNetValidation` occurs after validation has occurs. This time, we also have a reference to the result of the validation. Here we can do some additional processing on the error messages before they’re added to Model.
+      private ValidationFailure ConstructFailure(ValidationFailure failure)
+      {
+         var errorDetail = new
+         {
+             ErrorCode = failure.ErrorCode,
+             ErrorMessage = failure.ErrorMessage,
+             CustomState = failure.CustomState
+         };
+         var errorDetailString = JsonSerializer.Serialize(errorDetail, _jsonOptions);
+         failure.ErrorMessage = errorDetailString;
+         return failure;
+      }
+  
+      public ValidationResult AfterAspNetValidation(ActionContext actionContext,
+      IValidationContext validationContext,ValidationResult result)
+      {
+         if (result.Errors.Any())
+         {
+             var projection = result.Errors.Select(failure => ConstructFailure(failure));
+             return new ValidationResult(projection);
+         }
+
+         return result;
+      }
+```
+The first method `ConstructFailure` constructs the error message that would be provided if there is a validation error. 
+The second method `AfterAspNetValidation` displays the error message when there is a validation error.
```

Comment:
Still not quite right (and I may be partly to blame here as I didn't explain fully the first time).
The `UseErrorCodeInterceptor` class is used under the covers to make sure that any error code that is specified on a validation rule is actually included in the HTTP response. By default it just includes the error message and any code specified is left out. Clients should not need to invoke this class directly.

So when a validation rule does this...
```
RuleFor(x => x.ReasonForTermination).NotXssString()
             .WithErrorCode(ErrorCodes.XssCheckFailure);
```
then the error code value for `XssCheckFailure` will actually be included in the error details in the HTTP response.